### ISSUE TITLE
Frequent SIGSEGV in DBImpl::GetAggregatedIntProperty() because of a race condition

### ISSUE THREAD

**Issue by igorcanadi** [facebook/rocksdb#8440]
### Expected behavior
DBImpl::GetAggregatedIntProperty() doesn't crash.

### Actual behavior
DBImpl::GetAggregatedIntProperty() crashes. Stack trace:
```
#0  0x00007fc56d4134b2 in rocksdb::ThreadLocalPtr::Swap (this=0x0, ptr=0x7fc56d7da960 <rocksdb::SuperVersion::dummy>) at util/thread_local.cc:539
#1  0x00007fc56d133dda in rocksdb::ColumnFamilyData::GetThreadLocalSuperVersion (this=this@entry=0x7fc398c15800, db=db@entry=0x7fc4310fc800) at db/column_family.cc:1185
#2  0x00007fc56d17417e in rocksdb::DBImpl::GetAndRefSuperVersion (cfd=0x7fc398c15800, this=0x7fc4310fc800) at db/db_impl/db_impl.cc:3260
#3  rocksdb::DBImpl::GetIntPropertyInternal (this=0x7fc4310fc800, cfd=0x7fc398c15800, property_info=..., is_locked=<optimized out>, value=0x7fc55b374180) at db/db_impl/db_impl.cc:3188
#4  0x00007fc56d174283 in rocksdb::DBImpl::GetAggregatedIntProperty (this=0x7fc4310fc800, property=..., aggregated_value=0x7fc55b3741e0) at db/db_impl/db_impl.cc:3242
```

`local_sv_` is `nullptr` here: https://github.com/facebook/rocksdb/blob/065bea158798f5d67eb95ce339f3650db174c109/db/column_family.cc#L1200

I think the root cause is unlocking the DB mutex in `ColumnFamilyData::UnrefAndTryDelete()`, here: https://github.com/facebook/rocksdb/blob/065bea158798f5d67eb95ce339f3650db174c109/db/column_family.cc#L688

This code block executes when remaining ref count is 1, and it is correctly assumed that the ref count comes from the `super_version_` (see here: https://github.com/facebook/rocksdb/blob/065bea158798f5d67eb95ce339f3650db174c109/db/column_family.cc#L683). However, if we release the mutex another piece of code can grab the new ref, as `GetAggregatedIntProperty()` does. As I understand it, there is an implicit assumption that `local_sv_` is non-nullptr if a ref is non-zero.

This crashes our server process every 30min on a workload that drops column families every 30 seconds.

**Comment by ltamasi** [comment]
@igorcanadi Thanks for the detailed report! Hopefully this got fixed by #8605 but it would be great if you could confirm.

**Comment by igorcanadi** [comment]
Thanks @ltamasi . I will try this out in my next stress test, although the [workaround](https://github.com/rockset/rocksdb-cloud/commit/d9a050e05f1cb69055162bd5f2eaef93e86f252a) is working well for us.

**Comment by ltamasi** [comment]
Assuming this is fixed now (#8605 ); please feel free to reopen if this is not the case.
